tests/contrib/test_django_orm.py

```diff
@@ -138,27 +141,111 @@ def setUp(self):
-        self.assertEqual(storage.key_name, key_name)
-        self.assertEqual(storage.key_value, key_value)
-        self.assertEqual(storage.property_name, property_name)
+        self.assertEqual(storage.key_name, self.key_name)
+        self.assertEqual(storage.key_value, self.key_value)
+        self.assertEqual(storage.property_name, self.property_name)
+
+    @mock.patch("django.db.models")
+    def test_locked_get(self, djangoModel):
+        fake_model_with_credentials = FakeCredentialsModelMock()
+        entities = [
+            fake_model_with_credentials
+        ]
+        filter_mock = MagicMock(return_value=entities)
+        object_mock = MagicMock()
+        object_mock.filter = filter_mock
+        FakeCredentialsModelMock.objects = object_mock
+
+        storage = Storage(FakeCredentialsModelMock, self.key_name,
+                          self.key_value, self.property_name)
+        credential = storage.locked_get()
+        self.assertEqual(
+            credential, fake_model_with_credentials.credentials)
+
+    @mock.patch("django.db.models")
+    def test_locked_put(self, djangoModel):
+        storage = Storage(FakeCredentialsModelMock, self.key_name,
+                          self.key_value, self.property_name)
+        storage.locked_put(self.credentials)
+
+    @mock.patch("django.db.models")
+    def test_locked_put_with_overwite(self, djangoModel):
+        get_or_create_mock = MagicMock()
+        fake_credentials = FakeCredentialsModelMock()
+        get_or_create_mock.return_value = (fake_credentials, True)
+
+        object_mock = MagicMock()
+        object_mock.get_or_create = get_or_create_mock
+        FakeCredentialsModelMock.objects.get_or_create = get_or_create_mock
+
+        storage = Storage(FakeCredentialsModelMock, self.key_name,
+                          self.key_value, self.property_name)
+        storage.locked_put(self.credentials, True)
+        self.assertTrue(fake_credentials.saved)
+
+    @mock.patch("django.db.models")
+    def test_locked_delete(self, djangoModel):
+        fake_model_with_credentials = FakeCredentialsModelMock()
+        entities = [
+           fake_model_with_credentials
+        ]
+        filter_mock = MagicMock(return_value=entities)
+        object_mock = MagicMock()
+        object_mock.filter = filter_mock
+        FakeCredentialsModelMock.objects = object_mock
+
+
+class CredentialWithSetStore(CredentialsField):
+
+    def __init__(self):
+        self.model = CredentialWithSetStore
+
+    def set_store(self, storage):
+        pass
 
 
 class FakeCredentialsModel(models.Model):
+    """ FakeCredentials class that uses actual Django models methods.
+    """
+
+    def __init__(self):
+        self.deleted = False
+
     credentials = CredentialsField()
 
 
+class FakeCredentialsModelMock(object):
+    """ FakeCredentials class for when mocking out Django model methods is
+    necessary.
+    """
```

Comment:
This docstring is longer than 80 chars, but since in a test, no need for a docstring, just delete it so we don't have to fight it?